cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1436,6 +1440,32 @@ def make_resource(self, data, **kwargs):
         """Generate resource."""
         return SchedulerPluginSettings(**data)
 
+    @pre_load
+    def fetch_scheduler_definition(self, data, **kwargs):
+        """Fetch scheduler definition if it is s3 or https url."""
+        original_scheduler_definition = data["SchedulerDefinition"]
+        if type(original_scheduler_definition) is str:
+            try:
+                if original_scheduler_definition.startswith("s3"):
+                    bucket_parsing_result = parse_bucket_url(original_scheduler_definition)
+                    result = AWSApi.instance().s3.get_object(
+                        bucket_name=bucket_parsing_result["bucket_name"], key=bucket_parsing_result["object_key"]
+                    )
+                    data["SchedulerDefinition"] = yaml.safe_load(result["Body"].read().decode("utf-8"))
+                elif original_scheduler_definition.startswith("https"):
+                    with urlopen(original_scheduler_definition) as f:
+                        data["SchedulerDefinition"] = yaml.safe_load(f.read().decode("utf-8"))
+                else:
+                    raise ValidationError(
+                        f"SchedulerDefinition {data['SchedulerDefinition']} is invalid, "
```

Comment:
nit: what about "The provided value for SchedulerDefinition ('{url}') is invalid. You can specify this as an S3 URL, HTTPS URL or as an inline YAML object."

Reply:
done